lib/district.rb

```diff
@@ -1,41 +1,18 @@
-require_relative 'enrollment'
-require 'bigdecimal'
+require 'pry'
+
 class District
   attr_reader :name
+  attr_accessor :enrollment_data
 
   def initialize(name)
     @name = name[:name]
-
-  end
-
-  def enrollment
-    @enrollment_data
-  end
-
-  def get_enrollment(data)
-    @enrollment_data = data
   end
 
-  def calculate_kinder_average
-    enrollment_percentages = enrollment.kindergarten_participation.values
-
-    total_enrollment = enrollment_percentages.map { |percent| percent.to_f  }.reduce(:+)
-
-    average_district_percentage = total_enrollment/enrollment_percentages.count
-    average_district_percentage.round(3)
-   end
-
-  def calculate_hs_grad_average
-    if enrollment.school_data[:high_school_graduation_rates] == nil
-      grad_rates = 0.5
-    else
-      grad_rates = enrollment.high_school_graduation_rates.values
-
-      total_grad_rate = grad_rates.map {|grad_rate| grad_rate.to_f }.reduce(:+)
-
-      average_grad_rate = total_grad_rate/grad_rates.count
+  def calculate_field_average(enrollment_key)
```

Comment:
This method was consolidated from methods discerning between multiple high school/kindergarten averaging methods. This pattern continues throughout the application and violates DRY.